api/all/src/main/java/io/opentelemetry/api/internal/StringUtils.java

```diff
@@ -27,6 +27,7 @@
    * @param string a string reference to check
    * @return {@code true} if the string is null or is the empty string
    */
+  @Contract("null -> true")
```

Comment:
Is the intention that we only use this on `internal` code? I'm a little nervous about owning a public `Contract` annotation in this project.

Reply:
Nah we'd use it on any code not just this package - that's why it's in the `internal` folder, that it's only for us.

Reply:
But if we put it into our own public APIs, then we expose it to end-users, and who knows what they're going to do with it that we might break.

Reply:
Ah I see what you're saying. In that case I guess I'll say the intention is for internal code for now - more accurately I want to use it in ways to get nullaway checking our repo, for now this method is the only one I've found that needs it not sure what else will crop up later.

Reply:
I wonder if japicmp will pick up public usages of annotations. Can you verify that it does?

Reply:
It seems to detect if the `RetentionPolicy` is `RUNTIME` but not the current `CLASS`. There isn't really any use for it to be at runtime for us, but given we'll use it so infrequently I think the cost isn't a big deal either. Should I switch it to RUNTIME?

Reply:
Oh. If it's not RUNTIME we don't expose it to the resulting class files, right? So that should be ok to use even in non-internal classes, maybe?

Reply:
CLASS means only in the byte code at compile time (mostly annotation processor) but not when actually running the app on JVM. RUNTIME means also loaded during the running of the app, which has a tiny bit of memory usage. Both sort of expose the annotation to users, though CLASS is naturally much harder to actually use. I think either is fine here.

Reply:
could we use SOURCE here and thus not have them get into the bytecode at all? I don't know enough about how nullaway works to know if that would be enough.

Reply:
Oh duh - think that was my intention anyways but forgot while copying over from Jetbrains. SOURCE works and shouldn't have any caveats related to leaking out.